### PR TITLE
feat : added Custom posthog events in backend

### PR DESCRIPTION
**Pull request by Nazi-pikachu**
Added custom posthog events in backend for following routes
- `Update Link`
-  `Create Link`
- `Redirect Link`

### CODE REVIEW

**Comment by Nazi-pikachu** [comment]
@yuvrajsab 
This PR is now completed , it has all the changes you suggested
Please review this

File: apps/api/src/app/app.controller.ts [hunk 1]
```diff
@@ -101,6 +105,17 @@ export class AppController {
...
       })
       .subscribe();
+
+      this.telemetryService.sendEvent(
+        this.configService.get<string>("POSTHOG_DISTINCT_KEY"),
+        `Redirected Link`,
+        {
+          linkId:redirectedLink.id,
+          routeName: `/${hashid}}`,
+          routeParam: redirectedLink?.params,
```
Comment (yuvrajsab) [inline apps/api/src/app/app.controller.ts:115]: queryParams not routeParams

**Review by yuvrajsab — CHANGES_REQUESTED** [review]

File: apps/api/src/app/app.controller.ts (hunk 1 above)
Comment (Nazi-pikachu) [reply apps/api/src/app/app.controller.ts]: resolved

File: apps/api/src/app/app.controller.ts [hunk 2]
```diff
@@ -101,6 +105,18 @@ export class AppController {
...
       })
       .subscribe();
+
+      this.telemetryService.sendEvent(
+        this.configService.get<string>("POSTHOG_DISTINCT_KEY"),
+        `Redirected Link`,
+        {
+          linkId:redirectedLink.id,
+          routeName: `/${hashid}}`,
+          queryParams : redirectedLink?.params,
```
Comment (yuvrajsab) [inline apps/api/src/app/app.controller.ts:115]: this isn't what I meant by `params`. I was referring to the query params that are coming from the redirect url.

**Review by yuvrajsab — CHANGES_REQUESTED** [review]

File: apps/api/src/app/app.controller.ts (hunk 2 above)
Comment (yuvrajsab) [reply apps/api/src/app/app.controller.ts:115]: example: 
redirectUrl: `localhost:3000?abc=123&def=456`
```
queryParams: {
   abc: 123,
   def: 456
}
```

Comment (yuvrajsab) [reply apps/api/src/app/app.controller.ts:115]: good point. for now lets just remove queryparam field from here. we will see later what needs to be done.

Comment (Nazi-pikachu) [reply apps/api/src/app/app.controller.ts]: @yuvrajsab i have remove it

**Comment by Nazi-pikachu** [comment]
@yuvrajsab 
Can you please review this one and let me know if there is still any changes left to made in this PR

**Review by yuvrajsab — APPROVED** [review]
lgtm.

**Pull request merged**